samples/ck-rx65n/basic-sample/src/iotconnect_app.c

```diff
@@ -0,0 +1,267 @@
+    if (NULL != url) {
+        printf("Download URL is: %s\r\n", url);
+        const char *version = iotcl_clone_sw_version(data);
+        if (!version) {
+            printf("Failed to clone SW version! Out of memory?");
+            message = "Failed to clone SW version";
+        } else if (is_app_version_same_as_ota(version)) {
+            printf("OTA request for same version %s. Sending success\r\n", version);
+            success = true;
+            message = "Version is matching";
+        } else if (app_needs_ota_update(version)) {
+            printf("OTA update is required for version %s.\r\n", version);
+			success = false;
+			message = "Not implemented";
+        } else {
+            printf("Device firmware version %s is newer than OTA version %s. Sending failure\r\n", APP_VERSION,
+                    version);
+            // Not sure what to do here. The app version is better than OTA version.
+            // Probably a development version, so return failure?
+            // The user should decide here.
+            success = false;
+            message = "Device firmware version is newer";
+        }
+
+        free((void*) url);
+        free((void*) version);
+    } else {
+        // compatibility with older events
+        // This app does not support FOTA with older back ends, but the user can add the functionality
+        const char *command = iotcl_clone_command(data);
+        if (NULL != command) {
+            // URL will be inside the command
+            printf("Command is: %s\r\n", command);
+            message = "Old back end URLS are not supported by the app";
+            free((void*) command);
+        }
+    }
+    const char *ack = iotcl_create_ack_string_and_destroy_event(data, success, message);
+    if (NULL != ack) {
+        printf("Sent OTA ack: %s\r\n", ack);
+        iotconnect_sdk_send_packet(ack);
+        free((void*) ack);
+    }
+    if (needs_ota_commit) {
+        printf("Waiting for ack to be sent by the network\r\n.,,");
+        tx_thread_sleep(5 * NX_IP_PERIODIC_RATE);
+        UINT status = iotc_ota_fw_apply();
+        if (status) {
+            printf("Failed to apply firmware! Error was: %d\r\n", status);
+        }
+    }
+}
+
+static void command_status(IotclEventData data, bool status, const char *command_name, const char *message) {
+    const char *ack = iotcl_create_ack_string_and_destroy_event(data, status, message);
+    printf("command: %s status=%s: %s\r\n", command_name, status ? "OK" : "Failed", message);
+    printf("Sent CMD ack: %s\r\n", ack);
+    iotconnect_sdk_send_packet(ack);
+    free((void*) ack);
+}
+
+static void on_command(IotclEventData data) {
+	char* context = NULL;
+    char *command = iotcl_clone_command(data);
+    if (NULL != command) {
+    	char* token = strtok_r(command, " ", &context);
+    	if(!strncmp(token, "led", 3)) {
+    		token = strtok_r(NULL, " ", &context);
+    		RX65N_LED_STATE state = (token[0] == '1') ||

```

Comment:
A bit rusty on strok, but what if there is no argument? Missing null check?

Reply:
If there is no argument it will turn the LED off